dgl_sparse/src/sparse_matrix.cc

```diff
@@ -122,6 +124,22 @@ c10::intrusive_ptr<SparseMatrix> SparseMatrix::FromDiag(
   return SparseMatrix::FromDiagPointer(diag, value, shape);
 }
 
+c10::intrusive_ptr<SparseMatrix> SparseMatrix::RowwiseSelect(torch::Tensor ids) {
+  auto id_array = TorchTensorToDGLArray(ids);
+  auto new_shape = std::vector<int64_t>{id_array.GetSize(), this->shape()[1]};
+  if (HasCSR() || !HasCOO()) {
+    auto dgl_csr = CSRToOldDGLCSR(this->CSRPtr());
+    auto slice_csr = dgl::aten::CSRSliceRows(dgl_csr, id_array);
+    auto slice_value = DGLArrayToTorchTensor(dgl::aten::IndexSelect(TorchTensorToDGLArray(this->value()), slice_csr.data));
+    return SparseMatrix::FromCSRPointer(CSRFromOldDGLCSR(slice_csr), slice_value, new_shape);
+  } else { // COO
+    auto dgl_coo = COOToOldDGLCOO(this->COOPtr());
+    auto slice_coo = dgl::aten::COOSliceRows(dgl_coo, id_array);
+    auto slice_value = DGLArrayToTorchTensor(dgl::aten::IndexSelect(TorchTensorToDGLArray(this->value()), slice_coo.data));
```

Comment:
Pls use the index_select operator of Pytorch.

Reply:
Applied.